### PR TITLE
RELATED: RAIL-2999 - Tiger insights filtering

### PR DESCRIPTION
**Pull request by lupko**
-  getInsights now loads all data from backend and does client side filtering (because it is not yet available on server)
-  Added generic infrastructure to obtain all pages of data from backend
-  Added InMemoryPaging implementation of PagedResource

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2778/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4410/ : FAILURE in 5m 58s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2786/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4417/ : FAILURE in 2m 13s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2788/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4419/ : FAILURE in 6m 25s

File: libs/api-client-tiger/src/metadataUtilities.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+     * `page` number will be added dynamically upon each page request.
+     *
+     * @param client - API client to use, this is required so that function can correctly bind 'this' for
+     *  the entitiesGet function
+     * @param entitiesGet - function to get pages list of entities
+     * @param params - parameters accepted by the function
+     * @param options - options accepted by the function
+     * @internal
+     */
+    public static getAllPagesOf = async <T extends MetadataGetEntitiesResult>(
```
Comment (kandl) [inline libs/api-client-tiger/src/metadataUtilities.ts:109]: What about type it like this (to preserve inference in each client)?
```
public static getAllPagesOf = async <
        T extends MetadataGetEntitiesResult,
        P extends MetadataGetEntitiesParams
    >(
        client: ITigerClient,
        entitiesGet: MetadataGetEntitiesFn<T, P>,
        params: P,
        options: MetadataGetEntitiesOptions = {},
    ): Promise<T[]> 
```

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/api-client-tiger/src/metadataUtilities.ts (hunk 1 above)
Comment (lupko) [reply libs/api-client-tiger/src/metadataUtilities.ts]: done

**Review by kandl — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2793/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4424/ : SUCCESS in 13m 45s

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1122

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/826/ : SUCCESS in 26s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1090/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4428/ : SUCCESS in 13m 35s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/196/ : SUCCESS in 21m 37s (non-voting)

**Pull request merged**